account-decoder-client-types/src/lib.rs

```diff
@@ -0,0 +1,99 @@
+//! Core RPC client types for solana-account-decoder
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#[cfg(feature = "zstd")]
+use std::io::Read;
+use {
+    base64::{prelude::BASE64_STANDARD, Engine},
+    core::str::FromStr,
+    serde_derive::{Deserialize, Serialize},
+    serde_json::Value,
+    solana_account::WritableAccount,
+    solana_pubkey::Pubkey,
+};
+pub mod token;
+
+/// A duplicate representation of an Account for pretty JSON serialization
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct UiAccount {
+    pub lamports: u64,
+    pub data: UiAccountData,
+    pub owner: String,
+    pub executable: bool,
+    pub rent_epoch: u64,
+    pub space: Option<u64>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", untagged)]
+pub enum UiAccountData {
+    LegacyBinary(String), // Legacy. Retained for RPC backwards compatibility
+    Json(ParsedAccount),
+    Binary(String, UiAccountEncoding),
+}
+
+impl UiAccountData {
+    /// Returns decoded account data in binary format if possible
+    pub fn decode(&self) -> Option<Vec<u8>> {
```

Comment:
Why not also make this one a free function, like its `encode` counterpart? Then you can drop the `bs_58` dependency from this crate, possibly more.

Reply:
oh because rpc-client still needs the `decode()` methods, so I could make them free functions but I'd still want them in this crate for now